src/Microsoft.AspNetCore.OData/ODataController.cs

```diff
@@ -51,5 +52,108 @@ protected virtual UpdatedODataResult<TEntity> Updated<TEntity>(TEntity entity)
+        {
+            return new NotFoundODataResult(odataError);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Unauthorized (401) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>A <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual UnauthorizedODataResult Unauthorized(string message)
+        {
+            return new UnauthorizedODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Unauthorized (401) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>A <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual UnauthorizedODataResult Unauthorized(ODataError odataError)
+        {
+            return new UnauthorizedODataResult(odataError);
+        }
+
+#if NETCOREAPP3_1 || NETCOREAPP2_1 || NETSTANDARD2_1
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Conflict (409) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>A <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual ConflictODataResult Conflict(string message)
+        {
+            return new ConflictODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Conflict (409) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>A <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual ConflictODataResult Conflict(ODataError odataError)
+        {
+            return new ConflictODataResult(odataError);
+        }
+#endif
+
+        /// <summary>
+        /// Creates a <see cref="ActionResult"/> that when executed will produce an <see cref="ODataError"/> response.
+        /// </summary>
+        /// <param name="errorCode">Http Error code.</param>
+        /// <param name="message">Http Error Message.</param>
+        /// <returns></returns>
+        protected virtual ODataErrorResult ODataErrorRequest(string errorCode, string message)
+        {
+            return new ODataErrorResult(errorCode, message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ActionResult"/> that when executed will produce an <see cref="ODataError"/> response.
+        /// </summary>
+        /// <param name="odataError"><see cref="ODataError"/>.</param>
+        /// <returns></returns>
+        protected virtual ODataErrorResult ODataErrorResponse(ODataError odataError)
```

Comment:
>ODataErrorResponse [](start = 43, length = 18)

You have here an `ODataErrorResponse` and the previous method is called `ODataErrorRequest`. Shouldn't they both be called `ODataErrorResponse` as overloads of each others?

Reply:
Actually both should be `ODataErrorResult`
#Resolved